src/main/java/toughasnails/handler/thirst/FillBottleHandler.java

```diff
@@ -58,7 +58,7 @@ public void onRightClickCauldron(PlayerInteractEvent.RightClickBlock event)
         EntityPlayer player = event.getEntityPlayer();
         IBlockState state = world.getBlockState(event.getPos());
         
-        if (state.getBlock() instanceof BlockCauldron && SyncedConfig.getBooleanValue(GameplayOption.ENABLE_THIRST))
+        if (player.getHeldItem(event.getHand()).getItem() == Items.GLASS_BOTTLE && state.getBlock() instanceof BlockCauldron && SyncedConfig.getBooleanValue(GameplayOption.ENABLE_THIRST))
         {
```

Comment:
I guess it can be included with this, what I was saying though was that it's a separate fix that really should be split up from this